components/butterfly/src/server/mod.rs

```diff
@@ -71,12 +72,105 @@ pub trait Suitability: Debug + Send + Sync {
     fn get(&self, service_group: &ServiceGroup) -> u64;
 }
 
+/// Encapsulate a `Member` with the added understanding that this
+/// represents the identity of this Butterfly Server.
+///
+/// In particular, this localizes all incarnation increment and
+/// persistence logic.
+#[derive(Clone, Debug)]
+pub struct Myself {
+    member: Member,
+    // TODO (CM): This is only optional because the current
+    // implementation of Server requires it. See note there for more.
+    incarnation_store: Option<incarnation_store::IncarnationStore>,
```

Comment:
It seems to me the foundation of the butterfly server is poor. There's nothing wrong with a separate `new` and `start` functions, but that shouldn't leak into other structures where those distinctions aren't relevant. What need is there for an incarnation store in a non-started server?

There are a couple different ways we could approach this, but I think we should definitely refactor the existing code rather than continue to build on a shaky foundation and increase our tech debt.